blockchain/tx_pool_test.go

```diff
@@ -1973,6 +2048,150 @@ func TestTransactionReplacement(t *testing.T) {
+			tx: blobTransaction(0, 100000, big.NewInt(1000), big.NewInt(100), blobFeeCap, key, params.BlobTxMaxBlobs, func(sc *types.BlobTxSidecar) {
+				// Add an extra blob to sidecar but keep hashes the same
+				extraBlob := kzg4844.Blob{}
+				sc.Blobs = append(sc.Blobs, extraBlob)
+			}),
+			expectedError: fmt.Errorf("invalid number of %d blobs compared to %d blob hashes", params.BlobTxMaxBlobs+1, params.BlobTxMaxBlobs),
+		},
+		{
+			// Test rejection of blob transaction with invalid commitment number
+			name: "reject-blob-transaction-with-mismatched-commitment-number",
+			tx: blobTransaction(0, 100000, big.NewInt(1000), big.NewInt(100), blobFeeCap, key, params.BlobTxMaxBlobs, func(sc *types.BlobTxSidecar) {
+				// Add an extra commitment to sidecar but keep hashes the same
+				extraCommitment := kzg4844.Commitment{}
+				sc.Commitments = append(sc.Commitments, extraCommitment)
+			}),
+			expectedError: fmt.Errorf("invalid number of %d blob commitments compared to %d blob hashes", params.BlobTxMaxBlobs+1, params.BlobTxMaxBlobs),
+		},
+		{
+			// Test rejection of blob transaction with invalid commitment hash
+			name: "reject-blob-transaction-with-invalid-commitment-hash",
+			tx: blobTransaction(0, 100000, big.NewInt(1000), big.NewInt(100), blobFeeCap, key, params.BlobTxMaxBlobs, func(sc *types.BlobTxSidecar) {
+				// Modify the commitment to make hash mismatch
+				sc.Commitments[0][0] ^= 0xFF
+			}),
+			expectedError: errors.New("mismatches transaction"),
+		},
+		{
+			// Test rejection of blob transaction with invalid proof count
+			name: "reject-blob-transaction-with-invalid-proof-count",
+			tx: blobTransaction(0, 100000, big.NewInt(1000), big.NewInt(100), blobFeeCap, key, params.BlobTxMaxBlobs, func(sc *types.BlobTxSidecar) {
+				// Remove one proof to make count invalid
+				if len(sc.Proofs) > 0 {
+					sc.Proofs = sc.Proofs[:len(sc.Proofs)-1]
+				}
+			}),
+			expectedError: fmt.Errorf("invalid number of %d blob proofs expected %d", params.BlobTxMaxBlobs*kzg4844.CellProofsPerBlob-1, params.BlobTxMaxBlobs*kzg4844.CellProofsPerBlob),
+		},
+		{
+			// Test rejection of blob transaction with invalid cell proofs
+			name: "reject-blob-transaction-with-invalid-cell-proofs",
+			tx: blobTransaction(0, 100000, big.NewInt(1000), big.NewInt(100), blobFeeCap, key, 1, func(sc *types.BlobTxSidecar) {
+				// Corrupt the first proof
+				if len(sc.Proofs) > 0 {
+					sc.Proofs[0][0] ^= 0xFF
+				}
+			}),
+			expectedError: errors.New("invalid point encoding"),
+		},
+		{
+			// Test rejection of blob transaction with invalid sidecar version
+			name: "reject-blob-transaction-with-invalid-sidecar-version",
+			tx: blobTransaction(0, 100000, big.NewInt(1000), big.NewInt(100), blobFeeCap, key, 1, func(sc *types.BlobTxSidecar) {
+				sc.Version = types.BlobSidecarVersion0
+			}),
+			expectedError: fmt.Errorf("blob sidecar version %d not supported", types.BlobSidecarVersion0),
```

Comment:
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4844.md#networking
Is this the old network representative? I think its test would be better too.